app/src/main/java/swati4star/createpdf/util/PDFUtils.java

```diff
@@ -574,6 +578,39 @@ public boolean reorderRemovePDF(String inputPath, String output, String pages) {
         return outputPaths;
     }
 
+    private boolean isInputValid(String path, String[] ranges) {
+        try {
+            PdfReader reader = new PdfReader(path);
+            int numOfPages = reader.getNumberOfPages();
+            int startPage;
+            int endPage;
+
+
+            for (String range : ranges) {
+                if (!range.contains("-")) {
+                    startPage = Integer.parseInt(range);
+                    if (startPage > numOfPages) {
+                        showSnackbar(mContext, R.string.error_page_number);
+                        return false;
+                    }
+                } else {
+                    startPage = Integer.parseInt(range.substring(0, range.indexOf("-")));
+                    endPage = Integer.parseInt(range.substring(range.indexOf("-") + 1));
+                    if (startPage > numOfPages || endPage > numOfPages) {
+                        showSnackbar(mContext, R.string.error_page_number);
+                        return false;
+                    } else if (startPage >= endPage) {
+                        showSnackbar(mContext, R.string.error_page_range);
+                        return false;
+                    }
+                }
+            }
```

Comment:
From dry running on example `1-4 2-3`, I think this method will consider it valid but is it actually valid?

Reply:
why not? I think it's upto user to set how many pages he want in a single pdf.
1-4, 2-3 is a valid input.

Reply:
I think the input seems valid :+1:
You can leave it as is.